directory.cabal

```diff
@@ -36,31 +40,51 @@ source-repository head
 
 Library
     default-language: Haskell2010
+    default-extensions:
+        MultiWayIf
+        ViewPatterns
     other-extensions:
         CPP
         Trustworthy
 
     exposed-modules:
         System.Directory
+        System.Directory.AbstractFilePath
         System.Directory.Internal
+        System.Directory.Internal.AbstractFilePath
         System.Directory.Internal.Prelude
     other-modules:
         System.Directory.Internal.C_utimensat
+        System.Directory.Internal.Common.AbstractFilePath
         System.Directory.Internal.Common
         System.Directory.Internal.Config
+        System.Directory.Internal.Config.AbstractFilePath
         System.Directory.Internal.Posix
+        System.Directory.Internal.PosixFFI
+        System.Directory.Internal.Posix.AbstractFilePath
         System.Directory.Internal.Windows
+        System.Directory.Internal.WindowsFFI
+        System.Directory.Internal.WindowsFFI.Common
+        System.Directory.Internal.WindowsFFI.AbstractFilePath
+        System.Directory.Internal.Windows.AbstractFilePath
+        System.File.AbstractFilePath
+        System.File.PlatformFilePath
```

Comment:
These are inlined from https://github.com/hasufell/file-io ...I don't see another way currently.

Reply:
No, because it's not a boot library. We can't depend on non-boot libraries.

Reply:
This seems like a gap that needs to be resolved before any progress can be made? Reimplementing `file-io` capabilities seems beyond the scope of `directory`. If no other boot libraries has such capabilities then maybe `file-io` should be promoted to a boot library first?